datalake-api/src/main/java/com/sequenceiq/sdx/validation/UpgradeRequestValidator.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.sdx.validation;
+
+import java.util.Objects;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.sdx.api.model.SdxUpgradeRequest;
+
+public class UpgradeRequestValidator implements ConstraintValidator<ValidUpgradeRequest, SdxUpgradeRequest> {
+
+    @Override
+    public boolean isValid(SdxUpgradeRequest value, ConstraintValidatorContext context) {
+        if (Objects.isNull(value)) {
+            return false;
+        }
+        return isOsUpgrade(value) || isRuntimeUpgrade(value);
+    }
+
+    private boolean isOsUpgrade(SdxUpgradeRequest request) {
+        return Boolean.TRUE.equals(request.getLockComponents()) && StringUtils.isEmpty(request.getRuntime());
+    }
+
+    private boolean isRuntimeUpgrade(SdxUpgradeRequest request) {
+        return !Boolean.TRUE.equals(request.getLockComponents())
```

Comment:
!true = false

Reply:
This handles "null" value as Runtime upgrade